src/Mailgun/Api/Domain.php

```diff
@@ -0,0 +1,270 @@
+     */
+    public function listCredentials($domain, $limit = 100, $skip = 0)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::integer($limit);
+        Assert::integer($skip);
+
+        $params = [
+            'limit' => $limit,
+            'skip' => $skip,
+        ];
+
+        $response = $this->get(sprintf('/v3/domains/%s/credentials', $domain), $params);
+
+        return $this->safeDeserialize($response, CredentialListResponse::class);
+    }
+
+    /**
+     * Create a new SMTP credential pair for the specified domain.
+     *
+     * @param string $domain   Name of the domain.
+     * @param string $login    SMTP Username.
+     * @param string $password SMTP Password. Length min 5, max 32.
+     *
+     * @return Credential|array|ResponseInterface
+     */
+    public function newCredential($domain, $login, $password)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($login);
+        Assert::stringNotEmpty($password);
+        Assert::lengthBetween($password, 5, 32, 'SMTP password must be between 5 and 32 characters.');
+
+        $params = [
+            'login' => $login,
+            'password' => $password,
+        ];
+
+        $response = $this->postMultipart(sprintf('/v3/domains/%s/credentials', $domain), $params);
+
+        return $this->serializer->deserialize($response, SimpleResponse::class);
+    }
+
+    /**
+     * Update a set of SMTP credentials for the specified domain.
+     *
+     * @param string $domain Name of the domain.
+     * @param string $login  SMTP Username.
+     * @param string $pass   New SMTP Password. Length min 5, max 32.
+     *
+     * @return SimpleResponse|array|ResponseInterface
+     */
+    public function updateCredential($domain, $login, $pass)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($login);
+        Assert::stringNotEmpty($pass);
+        Assert::lengthBetween($pass, 5, 32, 'SMTP password must be between 5 and 32 characters.');
+
+        $params = [
+            'password' => $pass,
+        ];
+
+        $response = $this->putMultipart(
+            sprintf(
+                '/v3/domains/%s/credentials/%s',
+                $domain,
+                $login
+            ),
+            $params
+        );
+
+        return $this->serializer->deserialize($response, SimpleResponse::class);
+    }
+
+    /**
+     * Remove a set of SMTP credentials from the specified domain.
+     *
+     * @param string $domain Name of the domain.
+     * @param string $login  SMTP Username.
+     *
+     * @return SimpleResponse|array|ResponseInterface
+     */
+    public function deleteCredential($domain, $login)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($login);
+
+        $response = $this->delete(
+            sprintf(
+                '/v3/domains/%s/credentials/%s',
+                $domain,
+                $login
+            ),
+            []
```

Comment:
If we're going to send empty arrays, I would like to suggest to set an empty array by default for that parameter. It does not make sense to me write `[]` given it means "we are providing... (drumroll) Nothing!"